packages/input-selection/src/RoundRobinRandomImprove/change.ts

```diff
@@ -207,7 +207,21 @@ const pickExtraRandomUtxo = (
   return { utxoRemaining: newUtxoRemaining, utxoSelected: newUtxoSelected };
 };
 
-const coalesceChangeBundlesForMinCoinRequirement = (
+const mergeWithSmallestBundle = (values: Cardano.Value[], index: number): Cardano.Value[] => {
+  const toBeMerged = values.splice(index, 1)[0];
```

Comment:
This mixes up the styles: since we're returning the array, we shouldn't mutate the argument.
```suggestion
  const toBeMerged = [...values].splice(index, 1)[0];
```